app/components/post_textbox/post_textbox_base.js

```diff
@@ -408,62 +410,92 @@ export default class PostTextBoxBase extends PureComponent {
                 }],
             );
         } else {
+            const currentMembersCount = this.props.currentChannelMembersCount;
+            const notificationsToChannel = this.props.enableConfirmNotificationsToChannel;
+            const toAllorChannel = (/\B@(all|channel)\b/i).test(value);
```

Comment:
I suggests to pull out this regex test whether the text contains valid [at]all and [@]channel and have it tested with several test cases similar to how we're doing in the webapp - https://github.com/mattermost/mattermost-webapp/blob/master/utils/post_utils.test.jsx#L14

Reply:
Resolved!